src/main/java/download/IliasFileDownloaderTask.java

```diff
@@ -36,6 +36,10 @@ protected Void call() throws Exception {
 			entity = response.getEntity();
 
 			BufferedInputStream in = new BufferedInputStream(entity.getContent());
+			// remove invalid file endings
+			String targetPath = this.targetPath.endsWith(".")
+					? this.targetPath.substring(0, this.targetPath.length() - 1)
+					: this.targetPath;

```

Comment:
Good catch! What do you think about changing this in [IliasScraper](https://github.com/DeOldSax/iliasDownloaderTool/blob/2b5af55ff504ab71fe38f6a6e039aeb5deeaba23/src/main/java/control/IliasScraper.java#L177) directly when the file extension is scraped?

Reply:
You are much more familiar with the code base. If you think it should be included into the Scraper, you are most likely right. I just was in need of a quick fix to be able to bulk donwload from Ilias and wanted to share a working solution. To include this the right way, is too time consuming right now. Sorry. Maybe after my exams.